ibis/expr/tests/test_analysis.py

```diff
@@ -301,3 +301,14 @@ def test_join_table_choice():
     t = x.aggregate(cnt=x.n.count())
     predicate = t.cnt > 0
     assert L.sub_for(predicate, [(t, t.op().table)]).equals(predicate)
+
+
+def test_is_ancestor_analytic():
+    x = ibis.table(ibis.schema([('col', 'int32')]), 'x')
+    with_filter_col = x[x.columns + [ibis.null().name('filter')]]
+    filtered = with_filter_col[with_filter_col['filter'].isnull()]
+    subquery = filtered[filtered.columns]
+
+    with_analytic = subquery[subquery.columns + [subquery.count().name('analytic')]]
+
+    assert not subquery.op().is_ancestor(with_analytic)
```

Comment:
@maxmzkr The test I had in mind wasn't a test for `is_ancestor` since we're fairly certain that code doesn't have any effect. Constructing the correct SQL for `with_analytic` is what needs to be tested.